### ISSUE TITLE
xAxis min value doesn't update automatically on addPoint(), if new point x value is less then current xAxis min

### ISSUE THREAD

**Issue by vitaliy-kretsul** [highcharts/highcharts#15620]
#### Expected behaviour
On addPoint() xAxis min should update automatically, no matter if point x value is lower then current xAxis min

#### Actual behaviour
On first live demo attached, click on "Add Points" button several times, and you will see that navigator doesn't update it's selection.
But if the point has x value greater then current xAxis max, navigator updates correctly (second live demo).

#### Live demo with steps to reproduce
https://jsfiddle.net/Cortez/o9x2rtbe/12/ (point x value less then current xAxis min)
https://jsfiddle.net/Cortez/o9x2rtbe/14/ (point x value greater then current xAxis max)

#### Product version
Highstock 9.0.1

#### Affected browser(s)
Tested on Google Chrome

**Comment by raf18seb** [comment]
Hi @vitaliy-kretsul 

I wouldn't say it is a bug. You can simply set extremes to achieve the desired effect: https://jsfiddle.net/BlackLabel/n8u0sxfk/

**Comment by vitaliy-kretsul** [comment]
Hi @raf18seb 
Thanks for your reply. Yes, I am aware that I can use .setExtremes() for that. However, it feels counter-intuitive that I should do it.
Can you describe why exactly you consider it being not a bug?
Thank you

**Comment by raf18seb** [comment]
@vitaliy-kretsul what I meant is that it is working fine without any errors. The behavior is just inconsistent when adding data vs adding historical data.
What do you think @pawelfus?

**Comment by vitaliy-kretsul** [comment]
@raf18seb 
Yes, You are right. I would totally agree that it is more an inconsistency rather then bug.

**Comment by pawelfus** [comment]
I agree, it's inconsistent. In general, we let the navigator control the extremes when those are changed, but only if:
- a new point is added (at the end of the dataset) => `max` is higher than before -> increase the `xAxis.max` extreme
- an old point is removed (from the end of the dataset) => `min` is higher than before -> increase the `xAxis.min` extreme

**Workaround:**
Demo: https://jsfiddle.net/BlackLabel/f1wh9krt/

Plugin:
```js
(function(H) {
  H.wrap(
    H.Navigator.prototype,
    'updatedDataHandler',
    function(proceed, ...args) {
      const ret = proceed.apply(this, args),
        navigator = this.chart.navigator;

      navigator.stickToMin = navigator.stickToMin || (
        Math.round(navigator.zoomedMin) === 0 &&
        !this.chart.fixedRange
      )

      return ret;
    }
  )
})(Highcharts);
```

**Comment by raf18seb** [comment]
Because it's working correctly and there are 2 scenarios (default one and the second with a workaround) I am closing the ticket now. Feel free to leave your opinion on which one of those 2 behaviors should be the default one and open the ticket if needed.
